flink-runtime/src/main/java/org/apache/flink/runtime/deployment/TaskDeploymentDescriptorFactory.java

```diff
@@ -137,6 +148,16 @@ private List<InputGateDeploymentDescriptor> createInputGateDeploymentDescriptors
                                     consumedIntermediateResult, consumedPartitionGroup)));
         }
 
+        for (Map.Entry<IntermediateDataSetID, ShuffleDescriptor[]> entry :
+                consumedClusterPartitionShuffleDescriptors.entrySet()) {
+            inputGates.add(
+                    new InputGateDeploymentDescriptor(
+                            entry.getKey(),
+                            ResultPartitionType.BLOCKING_PERSISTENT,
+                            0,
```

Comment:
Why we always consumes subpartition 0? Does that mean for a cluster partition, each Task always produces one subpartition?

Reply:
Yes, the cluster partition producer should produces one subpartition.

Reply:
I'd suggest to explicitly add this assumption to the comment because it is not easy to understand unless one reads the assumption/limitation of the FLIP. Because theoretically in runtime, one partition can be consumed by multiple consumers.

Reply:
Comment of the assumption is added.